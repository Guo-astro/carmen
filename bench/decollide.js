var Benchmark = require('benchmark');
var suite = new Benchmark.Suite();
var assert = require('assert');
var termops = require('../lib/util/termops');
var fs = require('fs');
var UPDATE = process.env.UPDATE;

var expected = require('./expected/decollide');
var values = {};

suite
.add('decollide (collision)', function() {
    assert.equal(termops.decollide([], {
        properties: { 'carmen:text': 'r ademar da silva neiva #'}
    }, 'av francisco de aguirre # la serena'), false);
})
.add('decollide (clean)', function() {
    assert.equal(termops.decollide([], {
        properties: { 'carmen:text': 'av francisco de aguirre #'}
    }, 'av francisco de aguirre'), true);
})
.on('cycle', function(event) {
    var name = event.target.name;
    values[name] = event.target.hz;
    console.log(String(event.target));
    var compare = values[name]/expected[name];
    assert.ok(compare > 0.85);
    if (compare >= 1) console.log('  ', ((compare - 1) * 100).toFixed(2) + '%', 'faster than previously recorded');
    else console.log('  ', ((1 - compare) * 100).toFixed(2) + '%', 'slower than previously recorded');

})
.on('complete', function() {
    if (UPDATE) fs.writeFileSync(__dirname+'/expected/decollide.json', JSON.stringify(values, null, 2));
})
.run();