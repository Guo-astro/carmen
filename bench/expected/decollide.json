{
  "decollide (collision)": 424270.84763049387,
  "decollide (clean)": 427495.1103499325
}